docs: architecture with the DevWorkspace engine (#2125)
codenvy/che-docs: modules/administration-guide/partials/con_architecture-overview-with-devworkspace.adoc

[id="{prod-id-short}-architecture-overview-with-devworkspace{context}"]
= {prod-short} architecture with {devworkspace}

include::example$snip_using-devworkspace-warning.adoc[]

.High-level {prod-short} architecture with the {devworkspace} engine
image::architecture/che-interacting-with-devworkspace.png[]

> When {prod-short} is running with the {devworkspace} engine, it runs on three groups of components. The {prod-short} server components and the {devworkspace} engine are the two central server-side components of {prod-short}. Together they manage the User workspaces, which are container-based development environments.

Reviewer: You say that "it runs on three groups of components" but the next sentence mentions only two server-side components. What's the third component? Consider adding a summary list of the 3 components before describing the 2 server-side components. Maybe something like this (revise for accuracy about the 3rd component):

`When {prod-short} is running with the {devworkspace} engine, it runs on three groups of components: two central server-side components and [client-side?] User workspaces.`

Author: The third one: user workspaces.

Author: fixed

The {prod-short} dashboard controls the desired state of the User workspaces and displays the running workspaces.

The {devworkspace} engine manages the running state of the User workspaces.
The {devworkspace} engine stops a workspace after 30 minutes of inactivity. This duration is configurable.

> The User dashboard and the {devworkspace} engine communicate with valid {orch-name} objects: {devworkspace} Custom Resources.

Reviewer: Is `{devworkspace} Custom Resources` just the Che name for `{orch-name} objects`? Or are the custom resources a subset of K8s objects? If they are the same thing with a different name in Che, consider revising as follows: `communicate with valid {orch-name} objects, which are referred to as {devworkspace} Custom Resources in Che`.

In any case, revise somehow to replace the colon, which caused confusion for me.

Author: https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/

[id="{prod-id-short}-architecture-overview-with-devworkspace{context}"]
= {prod-short} architecture with {devworkspace}

include::example$snip_using-devworkspace-warning.adoc[]

.High-level {prod-short} architecture with the {devworkspace} operator
image::architecture/che-interacting-with-devworkspace.png[]

When {prod-short} is running with the {devworkspace} operator, it runs on three groups of components:

{prod-short} server components::

> Control the desired state of the User workspaces. The main component is the User dashboard, which displays the running workspaces.

Reviewer 2: ```suggestion
Manage User {orch-namespace} and workspaces. The main component is the User dashboard, from which users control their workspaces.
```
Not sure why User is with capital U but I have left as is.

{devworkspace} operator::

> Manages the running state of the User workspaces.

Reviewer 2: ```suggestion
Creates and controls the necessary {orch-name} objects to run User workspaces. Including `Pods`, `Services`, and `PeristentVolumes`.
```

> The {devworkspace} operator stops a workspace after 30 minutes of inactivity. This duration is configurable.

Reviewer 2: ```suggestion
```
It isn't relevant here. And it's not correct (currently a Che-Theia plugin idles Che-Theia workspaces)

User workspaces:: 

> Container-based development environments.

Reviewer 2: ```suggestion
Container-based development environments, the IDE included.
```
